src/pages/cart.jsx

```diff
@@ -23,13 +23,97 @@ import {
 import { Seo } from "../components/seo"
 
 export default function CartPage() {
-  const { checkout, loading } = React.useContext(StoreContext)
+  const { checkout, loading, client, setCheckout } = React.useContext(StoreContext)
   const emptyCart = checkout.lineItems.length === 0
 
-  const handleCheckout = () => {
+  const handleCheckout = async () => {
+    if(window.onwardApp) {
+      await window.onwardApp.updateCartForCheckout();
+    }
     window.open(checkout.webUrl)
   }
 
+  const parseCheckout = (checkout) => {
+    return {
+      currency: checkout.currencyCode || null,
+      items: checkout.lineItems.map(item =>
+        ({
+          id: item.id,
+          cost: { amount: Number(item.variant.price.amount) * item.quantity, currencyCode: item.variant.price.currencyCode },
+          quantity: item.quantity,
+          properties: null,
+          variant: {
+            price: null,
+            requiresShipping: true,
+            id: item.variant.id,
+          }
+        })
+      ) || []
+    };
+  };
+
+  React.useEffect(() => {
+    if(window.onwardApp) {
+      window.onwardApp.cartChanged(parseCheckout(checkout));
+    }
+  }, [checkout]);
+
+  React.useEffect(() => {
+    window.onwardStorefrontClient = {
```

Comment:
this is the meat of the client, I put it in a `useEffect` because that's the only place I had access to `window`. I think this could all be moved to it's own file if we wanted to as well.

Reply:
yes, but I just refactored things so it doesn't now and can be passed in to the `initalizeOnward` function. We still need to set up the onward settings on the window though.